leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -93,7 +107,7 @@ public RESPONSE call() throws Exception {
             while (acksRecieved < quorumSize) {
                 try {
                     // check if quorum is impossible (nack quorum failure)
-                    if (nacksRecieved > remotes.size() - quorumSize) {
+                    if (shortcircuitIfQuorumImpossible && nacksRecieved > remotes.size() - quorumSize) {
```

Comment:
You're right that this check is only for PaxosSynchronizer, but it's not just for the tests 🙂 In "normal" operation, currently if you get a nack from anyone, you'll immediately stop trying to get responses from the other nodes, which is not the intended behaviour.